### PR TITLE
feat: switch all object detection models to Triton (#622)

### LINKED ISSUE #942: Process crash when calling /predict/images with universal-logo-detector model
### What

API thread is most of the time killed (502 Bad Gateway) when `/predict/images` is called with `universal-logo-detector` or `nutrition-table` models.
https://robotoff.openfoodfacts.org/api/v1/images/predict?image_url=https://fr.openfoodfacts.org/images/products/801/973/005/1888/front_it.3.full.jpg&models=universal-logo-detector&output_image=1

The culprit is the size of the JSON response returned by TF Serving: 150 MB of text, 950 MB in RAM once deserialized.
In the response, we can find the features of the last layer of the network (which we don't need) that is really heavy.

This also has the downside of slowing down image processing and increasing memory usage in Robotoff workers.

Solution: export again the model as SavedModel without outputting this field, or find a way to query Tensorflow Serving to return predictions without this field.

### PR DESCRIPTION
**Pull request by raphael0202**
- switch to Triton for all object detection model, use gRPC protocol for faster transfer
- only request needed output fields (resolves #942)
- get rid of CategoryIndex and protobuf-generated files

I managed to migrate the category detector model to Triton (using the SavedModel asset) but didn't succeed in sending inference request, due to dimension issue: `ingredient` input shape is defined as (-1, ), `product_name` as (1, ). Triton requires all inputs to have the same number of elements for dim 0 (=batch_size), while it's not necessary in Tensorflow Serving as inputs are fed differently.
We could fix this dimension issue in the category predictor model code, but as we're going to change it anyway to a new model we can keep Tensorflow Serving to serve category prediction model in the meantime.

### CODE REVIEW

File: .github/workflows/models-deploy.yml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 name: Sync Tensorflow models to Proxmox VM
-on: 
+on:
   push:
     branches:
       - master
       - deploy-*
-    tags: 
+    tags:
       - v*.*.*
```
Comment (alexgarel) [inline .github/workflows/models-deploy.yml:8]: we have to change this to deploy on specific tags. see https://github.com/openfoodfacts/robotoff/issues/565

File: robotoff/prediction/object_detection/core.py [hunk 2]
```diff
@@ -156,18 +197,18 @@ def get_available_models(cls) -> List[str]:
 
     @classmethod
     def load_all(cls):
```
Comment (alexgarel) [inline robotoff/prediction/object_detection/core.py:199]: I see this method is called at module importation (I find it unperfect already !).
Why do we do that, if it is meant to be served by triton ?

File: docker/ml.yml [hunk 3]
```diff
@@ -14,5 +14,25 @@ services:
...
+    # - GRPC/HTTP support
+    # - CPU only (we don't have GPU in production)
+    # - Tensorflow 2 SavedModel and ONNX support
+    # This allows us to reduce significantly the image size
+    # See https://gist.github.com/raphael0202/091e521f2c79a8db8c6e9aceafb6e0b9 for build script
+    image: ghcr.io/openfoodfacts/triton:latest
+    ports:
+      - 8000:8000
+      - 8001:8001
+      - 8002:8002
```
Comment (alexgarel) [inline docker/ml.yml:29]: I really would prefer if all port exposure where configurable through .env. 

See https://github.com/openfoodfacts/openfoodfacts-infrastructure/blob/develop/docs/docker.md#enable-configuration-through-environment

```suggestion
      - ${TRITON_EXPOSE_HTTP:-8000}:8000
      - ${TRITON_EXPOSE_GRPC:-8001}:8001
      - ${TRITON_EXPOSE_METRICS:-8001}:8002
```
In this case, it also has the advantage of documenting what ports are doing :blush:

**Review by alexgarel — APPROVED** [review]
I have some small comments, so I already approve, but it would be great to do changes.

File: robotoff/prediction/object_detection/core.py (hunk 2 above)
Comment (raphael0202) [reply robotoff/prediction/object_detection/core.py:199]: We need the ID->label mapping (which is now in model directory in `labels.txt`), but it's indeed not necessary to load it at import time, we can load it lazily.

File: docker/ml.yml (hunk 3 above)
Comment (raphael0202) [reply docker/ml.yml:29]: It's a good practice indeed, I'm adding this

**Pull request merged**